docs/Developers/Linter_Bears_Advanced.rst

```diff
@@ -41,6 +40,28 @@ The path of the temporary configuration file can be accessed inside
         def create_arguments(filename, file, config_file):
             return "--use-config", config_file
 
+If you return ``None``, no configuration file is generated. A common case
+where to explicitly return ``None`` is when the user wants to use his own
+tool-specific config. If a config file is supplied by user that means we
+don't have to generate a config file. Generating config file consumes
+resources. Avoiding generating config file increases the performance.
+
+::
+
+    @linter(executable='...')
+    class MyBear:
+        def generate_config(filename, file, settingA: bool):
```

Comment:
maybe call it `user_config` to follow better the explanation above

Reply:
ah wait `user_config` has to be added together with `settingA` (which should be renamed to `setting_a` btw :smile:)